### DISCUSSION TITLE
Error: org.exoplatform.web.security.codec.CodecKeyChangedException when I use a Exo-docker version 6.5_lastest

### DISCUSSION THREAD

**Discussion by vitorpadilha** [exoplatform/exo-community#42] [Q&A]
Hello,

  I down the docker containter with exoplatform and this error appear when i try up the Application again.

This is the message: "Caused by: org.exoplatform.web.security.codec.CodecKeyChangedException: Symetric-Key /codec/codeckey.txt used to encrypt/decrypt sensitive data seems to be changed. This may lead to data inconsistency."

Can someone help-me.

Thanks.

**Comment by rdenarie** [comment]
Hello,

Do you have data you want to keep in the platform ? 

In fact, the file codeckey.txt is a file used to init so element of the platform. It is for example use by the password encoder to encode user password in database. So if this file change, password cannot be validated.
During the down, your container where removed, and so the codeckey.txt file as it is not mounted in a volume.

So when you re-up the server, the file was recreated, generating the error.

So there is two options : 
- no critical data : down containers, drop all related volumes, and restart. All the platform will be recreated. It will erase all existing data.
- Data to conserv : I have some steps to test locally before giving you answers.

So, please, say me if the first solution is ok for you not. If no, I'll dig into the problem.

In the same time, I will update the docker-compose file to externalize the codeckey.txt, so that the problem will no be reproduced in the future.
